apollo-federation-types/src/build_plugin/build_message.rs

```diff
@@ -0,0 +1,137 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+#[non_exhaustive]
+pub enum BuildMessageLevel {
+    Debug,
+    Info,
+    Warn,
+    Error,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct BuildMessageLocation {
```

Comment:
should add a comment to all of these structs that any new fields must be wrapped in an `Option` type in order to maintain backwards compatibility. this is to allow new versions of rover to update this struct and still be able to read output from plugins that don't include the optional field. if these structs add a new required field, old plugins won't be able to run anymore because their output won't include that new required field.